vpr/src/pack/prepack.cpp

```diff
@@ -742,65 +817,75 @@ t_pack_molecule *alloc_and_load_pack_molecules(
         std::multimap<AtomBlockId,t_pack_molecule*>& atom_molecules,
         std::unordered_map<AtomBlockId,t_pb_graph_node*>& expected_lowest_cost_pb_gnode,
 		const int num_packing_patterns) {
-	int i, j, best_pattern;
 	t_pack_molecule *list_of_molecules_head;
 	t_pack_molecule *cur_molecule;
-	bool *is_used;
-    auto& atom_ctx = g_vpr_ctx.atom();
-
-	is_used = (bool*)vtr::calloc(num_packing_patterns, sizeof(bool));
+	auto& atom_ctx = g_vpr_ctx.atom();
+	int best_pattern;
 
 	cur_molecule = list_of_molecules_head = nullptr;
 
-	/* Find forced pack patterns
-	 * Simplifying assumptions: Each atom can map to at most one molecule,
-     *                          use first-fit mapping based on priority of pattern
-	 * TODO: Need to investigate better mapping strategies than first-fit
-     */
-	for (i = 0; i < num_packing_patterns; i++) {
-		best_pattern = 0;
-		for(j = 1; j < num_packing_patterns; j++) {
-			if(is_used[best_pattern]) {
-				best_pattern = j;
-			} else if (is_used[j] == false && compare_pack_pattern(&list_of_pack_patterns[j], &list_of_pack_patterns[best_pattern]) == 1) {
-				best_pattern = j;
-			}
-		}
-		VTR_ASSERT(is_used[best_pattern] == false);
-		is_used[best_pattern] = true;
+	bool *feasible_patterns = new bool[num_packing_patterns];
```

Comment:
Use std::vector<uint8> here, over raw new and free?

Reply:
done